packages/ui/src/components/form-primitives/form-wrapper.tsx

```diff
@@ -27,11 +27,12 @@ export function FormWrapper<T extends FieldValues>({
   formRef,
   formState,
   control,
+  onSubmit,
   ...props
 }: FormWrapperProps<T>) {
   return (
     <FormProvider {...props} formState={formState} control={control}>
-      <form className={cn('cn-form', className)} ref={formRef} {...props} noValidate>
+      <form className={cn('cn-form', className)} ref={formRef} onSubmit={onSubmit} noValidate>
```

Comment:
This fixes this type of error. Currently, we have a bunch of them in the console
<img width="390" alt="image" src="https://github.com/user-attachments/assets/e865829b-26c1-444b-98be-07734d8db5fc" />

Reply:
NIT: I am aware of this issue. I have it in my backlog. Do we need to spread props in `form` element for any other use case?

Reply:
I'm pretty sure we don't need it. `<form>` only needs the onSubmit callback
https://react-hook-form.com/docs/formprovider